tests/functional/object/mcg/test_bucket_replication.py

```diff
@@ -276,32 +276,41 @@ def test_bidirectional_bucket_replication(
         """
 
         first_bucket_name = bucket_factory(bucketclass=first_bucketclass)[0].name
-        replication_policy = ("basic-replication-rule", first_bucket_name, None)
-        second_bucket_name = bucket_factory(
-            1, bucketclass=second_bucketclass, replication_policy=replication_policy
-        )[0].name
+        second_bucket_name = bucket_factory(bucketclass=second_bucketclass)[0].name
+        prefix_site1 = "site1"
+        prefix_site2 = "site2"
 
         patch_replication_policy_to_bucket(
-            first_bucket_name, "basic-replication-rule-2", second_bucket_name
+            second_bucket_name,
+            "basic-replication-rule-1",
+            first_bucket_name,
+            prefix_site2 + "/",
         )
 
-        standard_test_obj_list = awscli_pod_session.exec_cmd_on_pod(
-            f"ls -A1 {AWSCLI_TEST_OBJ_DIR}"
-        ).split(" ")
+        patch_replication_policy_to_bucket(
+            first_bucket_name,
+            "basic-replication-rule-2",
+            second_bucket_name,
+            prefix_site1 + "/",
+        )
 
         # Write all downloaded objects to the bucket
-        sync_object_directory(
+        written_objects = write_random_test_objects_to_bucket(
             awscli_pod_session,
-            AWSCLI_TEST_OBJ_DIR,
-            f"s3://{first_bucket_name}",
-            mcg_obj_session,
+            first_bucket_name,
+            test_directory_setup.origin_dir,
+            amount=10,
+            prefix=prefix_site1,
+            mcg_obj=mcg_obj_session,
         )
-        first_bucket_set = set(standard_test_obj_list)
-        assert first_bucket_set == {
+        written_objects_with_prefix = list(
+            map(lambda x: prefix_site1 + "/" + x, written_objects)
+        )
```

Comment:
The use of `map` is clever but a bit unreadable IMO. How about this for simplicity?
```suggestion
        written_objects_with_prefix = [f"{prefix_site1}/{obj}" for obj in written_objects]
```

Reply:
Yes we can use both logic. I will go ahead and add for loop for simplicity